frontend/src/components/data-table/cell-selection/types.ts

```diff
@@ -0,0 +1,51 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import type { OnChangeFn, RowData, Updater } from "@tanstack/react-table";
+
+export type CellSelectionItem = { rowId: string; columnName: string };
+export type CellSelectionState = CellSelectionItem[];
+export type CellSelectionTableState = {
+  cellSelection: CellSelectionState;
+};
+
+export interface CellSelectionOptions {
+  enableCellSelection?: boolean;
+  enableMultiCellSelection?: boolean;
+  onCellSelectionChange?: OnChangeFn<CellSelectionState>; // | CellSelectionItem ???
+}
+
+export interface CellSelectionCell {
+  /**
+   * Returns whether or not the cell is selected.
+   */
+  getIsSelected: () => boolean;
+
+  /**
+   * Selects/deselects the cell.
+   */
+  toggleSelected: (value?: boolean) => void;
+}
+
+// These are additional properties add to the table instance in createTable
+export interface CellSelectionInstance<TData extends RowData> {
+  setCellSelection: (updater: Updater<CellSelectionState>) => void;
+
+  toggleAllRowsSelected: (value?: boolean) => void;
+
+  resetCellSelection(defaultValue?: CellSelectionState): void;
+
+  getIsAllCellsSelected: () => boolean;
+
+  getIsAllPageCellsSelected: () => boolean;
```

Comment:
can you make all these fields optional? its a have-to-know footgun of the table plugin system. since plugins are optional. the other tables may not have these methods and they could error (we use the table in the variable sidebar)